lib/src/utils/organize_imports/organize_imports_in_paths.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:io';
+
+import 'package:io/ansi.dart';
+
+import 'organize_imports.dart';
+
+/// Organizes imports in a list of files and directories.
+int organizeImportsInPaths(
+  Iterable<String> paths, {
+  bool check = false,
+  bool verbose = false,
+}) {
+  final allFiles = _getAllFiles(paths);
+  return _organizeImportsInFiles(allFiles, verbose: verbose, check: check);
+}
+
+/// Returns all file paths from a given set of files and directories.
+Iterable<String> _getAllFiles(Iterable<String> paths) {
+  final allFiles = Set<String>();
+
+  for (final path in paths) {
+    switch (FileSystemEntity.typeSync(path)) {
+      case FileSystemEntityType.directory:
+        // skip hidden directories
+        if (path.startsWith('.')) {
+          continue;
+        }
+        final children = Directory(path).listSync().map((e) => e.path);
+        allFiles.addAll(_getAllFiles(children));
+        break;
+      case FileSystemEntityType.file:
+        if (File(path).path.endsWith('.dart')) {
+          allFiles.add(path);
+        }
+        break;
+      case FileSystemEntityType.link:
+        // skip links
+        break;
+      default:
+        throw Exception('Unknown FileSystemEntity type encountered');
+    }
+  }
+
+  return allFiles;
+}
+
+/// Organizes imports in a list of files.
+int _organizeImportsInFiles(
+  Iterable<String> paths, {
+  bool verbose = false,
+  bool check = false,
+}) {
+  for (final path in paths) {
+    final exitCode =
+        _organizeImportsInFile(path, verbose: verbose, check: check);
+    if (exitCode != 0) {
+      return exitCode;
+    }
+  }
+  return 0;
```

Comment:
#nit not a big deal, but we could avoid returning early here so that everything that is valid still gets organized and the user receives an error message for each file that failed - this is actually what `dart format` does when there are syntax errors.

```suggestion
  var exitCode = 0;
  for (final path in paths) {
    final codeForFile =
        _organizeImportsInFile(path, verbose: verbose, check: check);
    if (codeForFile != 0) {
      exitCode = codeForFile;
    }
  }
  return exitCode;
```